### PR TITLE
Ctctraders 505 cache: Cache UserAnswers

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/controllers/UpdateRejectedMRNController.scala [hunk 1]
```diff
@@ -48,9 +51,13 @@ class UpdateRejectedMRNController @Inject()(override val messagesApi: MessagesAp
...
-        case Some((_, mrn)) =>
-          val json = Json.obj("form" -> form.fill(mrn), "arrivalId" -> arrivalId.value)
-          renderer.render("updateMovementReferenceNumber.njk", json).map(Ok(_))
+        case Some(arrivalMovementRequest) =>
+          MovementReferenceNumber(arrivalMovementRequest.header.movementReferenceNumber) match {
+            case Some(mrn) =>
+              val json = Json.obj("form" -> form.fill(mrn), "arrivalId" -> arrivalId.value)
+              renderer.render("updateMovementReferenceNumber.njk", json).map(Ok(_))
+            case _ => Future.successful(Redirect(routes.TechnicalDifficultiesController.onPageLoad()))
+          }
```
Comment (amaalali) [inline app/controllers/UpdateRejectedMRNController.scala:60]: This is a more stylistic comment, but mapping (and flatMapping) on the Options here as opposed to pattern matching will allow us to reduce the number of branches handing the negative cases and will make the code more compact. This isn't a require change though so feel free to ignore

File: app/services/UserAnswersService.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  def getUserAnswers(arrivalId: ArrivalId)(implicit hc: HeaderCarrier): Future[Option[UserAnswers]] =
+    arrivalNotificationMessageService
+      .getArrivalNotificationMessage(arrivalId)
+      .map(
+        _.flatMap {
+          arrivalMovementRequest =>
+            ArrivalMovementRequestConversionService.convertToArrivalNotification(arrivalMovementRequest).flatMap {
+              arrivalNotification =>
+                UserAnswersConversionService.convertToUserAnswers(arrivalNotification)
```
Comment (amaalali) [inline app/services/UserAnswersService.scala:36]: I think this bit was meant to be replaced by ArrivalMovementRequestToUserAnswersService? It separates out the concern of the conversion, and at the moment this service does a few things

File: test/generators/MessagesModelGenerators.scala [hunk 3]
```diff
@@ -232,7 +233,7 @@ trait MessagesModelGenerators extends Generators {
         messageSender               <- arbitrary[MessageSender]
         interchangeControlReference <- arbitrary[InterchangeControlReference]
         date                        <- arbitrary[LocalDate]
-        time                        <- arbitrary[LocalTime]
+        time                        <- Gen.const(LocalTime.of(14, 20))
```
Comment (amaalali) [inline test/generators/MessagesModelGenerators.scala:236]: Why are we fixing this value as opposed to using the LocalTime generators?

Comment (ashwinirudrappa) [reply test/generators/MessagesModelGenerators.scala:236]: @amaalali this is to fix that issue with seconds

File: app/services/UserAnswersService.scala (hunk 2 above)
Comment (ashwinirudrappa) [reply app/services/UserAnswersService.scala:36]: @amaalali Can we have a chat, I don't remember this bit I was not in that conversation

File: test/generators/MessagesModelGenerators.scala (hunk 3 above)
Comment (amaalali) [reply test/generators/MessagesModelGenerators.scala:236]: So we don't have second here, right, in that case we should take a similar approach to this: https://github.com/hmrc/transit-movements-trader-at-destination/blob/master/test/generators/JavaTimeGenerators.scala#L52-L57

File: app/controllers/UpdateRejectedMRNController.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/controllers/UpdateRejectedMRNController.scala:60]: @amaalali it's a little bit tricky, have to do a get or else to return the same difficulty page

**Pull request merged**